pkg/test/kube/apply.go

```diff
@@ -0,0 +1,322 @@
+func (a *accessorImpl) ApplyContentsDryRun(namespace string, contents string) ([]string, error) {
+	return a.applyContents(namespace, contents, true)
+}
+
+// Apply applies the config in the given filename using kubectl.
+func (a *accessorImpl) Apply(namespace string, filename string) error {
+	return a.apply(namespace, filename, false)
+}
+
+// ApplyDryRun applies the config in the given filename using kubectl with DryRun mode.
+func (a *accessorImpl) ApplyDryRun(namespace string, filename string) error {
+	return a.apply(namespace, filename, true)
+}
+
+// applyContents applies the given config contents using kubectl.
+func (a *accessorImpl) applyContents(namespace string, contents string, dryRun bool) ([]string, error) {
+	files, err := a.contentsToFileList(contents, "accessor_applyc")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := a.applyInternal(namespace, files, dryRun); err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+// apply the config in the given filename using kubectl.
+func (a *accessorImpl) apply(namespace string, filename string, dryRun bool) error {
+	files, err := a.fileToFileList(filename)
+	if err != nil {
+		return err
+	}
+
+	return a.applyInternal(namespace, files, dryRun)
+}
+
+func (a *accessorImpl) applyInternal(namespace string, files []string, dryRun bool) error {
+	s, err := a.getSchema()
+	if err != nil {
+		return err
+	}
+
+	for _, f := range removeEmptyFiles(files) {
+		if err := a.applyFile(namespace, f, s, dryRun); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *accessorImpl) applyFile(namespace string, file string, s *openAPISchema, dryRun bool) error {
+	if dryRun {
+		scopes.Framework.Infof("Applying YAML file (DryRun mode): %v", file)
+	} else {
+		scopes.Framework.Infof("Applying YAML file: %v", file)
+	}
+
+	namespaceToUse := "default"
+	enforceNamespace := false
+	if len(namespace) > 0 {
+		namespaceToUse = namespace
+		enforceNamespace = true
+	}
+	filenameOptions := &resource.FilenameOptions{
+		Filenames: []string{file},
+	}
+	r := resource.NewBuilder(a.clientGetter).
+		Unstructured().
+		Schema(newValidator(s)).
+		ContinueOnError().
+		NamespaceParam(namespaceToUse).DefaultNamespace().
+		FilenameParam(enforceNamespace, filenameOptions).
+		Flatten().
+		Do()
+	infos, err := r.Infos()
+	if err != nil {
+		return err
+	}
+	if len(infos) == 0 {
+		return fmt.Errorf("no objects to apply")
+	}
+
+	// Iterate through all objects, applying each one.
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+	for _, info := range infos {
+		if err := a.applyObject(info, s, dryRun, stdout, stderr); err != nil {
+			// Concatenate the stdout and stderr
+			s := stdout.String() + stderr.String()
+			scopes.Framework.Infof("Failed applying YAML file: %s (err: %v): %s", file, err, s)
+			return fmt.Errorf("%v: %s", err, s)
+		}
+	}
+	return nil
+}
+
+func (a *accessorImpl) applyObject(info *resource.Info, s *openAPISchema, dryRun bool, stdout io.Writer,
```

Comment:
This is not really maintainable... If we are going to try to literally be kubectl we should probably use the kubectl library. If we aren't then we can make things much simpler I think. I would expect we can just do 
```
cl.Patch(context.TODO(), us.GetName(), types.ApplyPatchType, bytes, metav1.PatchOptions{
                                FieldManager: "istio-testing",
                                Force:        util.BoolPointer(true),
                        })
```

and bypass most of this code